src/components/auth.js

```diff
@@ -1,61 +1,171 @@
-        try {
-            await signInWithEmailAndPassword(auth, email, password);
-            console.log("User successfully signed in")
-        } catch(err) {
-            console.error("Error signing in:", err);
-        }
-    };
-
-    const signUp = async () => {
-        try {
-            await createUserWithEmailAndPassword(auth, email, password);
-            console.log("Account successfully created")
-        } catch(err) {
-        
-            console.error("Error signing up:", err);
-  
-        }
-    };
-    
-    const logout = async () => {
-        try {
-            await signOut(auth);
-            console.log("User successfully logged out")
-        } catch(err) {
-            console.error(err);
-        }
-    };
-
-    return (
-   
-        <div>
-            <Input 
-            placeholder="Email..."
+import {
+  Button,
+  Input,
+  Box,
+  Flex,
+  Stack,
+  Link,
+  Text,
+  InputGroup,
+  InputRightElement,
+  Alert,
+  AlertIcon,
+} from '@chakra-ui/react';
+import {
+  createUserWithEmailAndPassword,
+  signOut,
+  signInWithEmailAndPassword,
+} from 'firebase/auth';
+import { useEffect, useState } from 'react';
+import { useNavigate, Link as RouterLink } from 'react-router-dom';
+
+const Auth = (props) => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [password2, setPassword2] = useState("");
+  const [show, setShow] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [passRules, setPassRules] = useState(false);
+  const isLoginPage = props.isLoginPage;
+  const navigate = useNavigate();
+
+  const handleClick = () => setShow(!show);
+
+  const signIn = async () => {
+    try {
+      await signInWithEmailAndPassword(auth, email, password);
+      navigate('/dashboard');
+      console.log("User successfully signed in");
+    } catch (err) {
+      setErrorMessage("Error signing in. Check either username or password.");
+      console.error("Error signing in:", err);
+    }
+  };
+
+  const signUp = async () => {
+    try {
+      await createUserWithEmailAndPassword(auth, email, password);
+      console.log("Account successfully created");
+      navigate('/dashboard');
+    } catch (err) {
+      console.error("Error signing up:", err);
+      setErrorMessage("Error signing up. Please try again.");
+    }
+  };
+
+  const goToForgetPassword = async () => {
+    navigate('/login/password-reset');
+  };
+
+  const logout = async () => {
+    try {
+      await signOut(auth);
+      console.log("User successfully logged out");
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  const checkPass = () => {
+    if (password !== password2) {
+      setErrorMessage("Passwords do not match");
+    } else {
+      setErrorMessage(""); // Clear any previous error message
+      signUp();
+    }
+  };
+
+
+  useEffect(() => {
+    if (password.length >= 6) {
+      setPassRules(false);
```

Comment:
Wait, so we don't pass the rules if our password is more than 6 characters?

Reply:
I had it so that while the password is less than 6 (which is the only password criteria) for it to show up. Once the password has met the criteria (greater than 6) the criteria message goes away.

Reply:
Oh I see. That's pretty confusing naming-wise since I'd expect that if passRules is true then I have passed the rules. It'd probably be easier to read if you just flipped all the booleans here, ie the default value is true, you set it to false if it's less than 6, etc.

Reply:
I have fixed the password issue